session_test.go

```diff
@@ -379,107 +343,110 @@ func TestClose_closeTimeout(t *testing.T) {
 }
 
 func TestNonNilInterface(t *testing.T) {
-	_, server := testClientServer()
+	_, server := testClientServer(t)
 	server.Close()
 
 	conn, err := server.Accept()
+	if err == nil || !errors.Is(err, ErrSessionShutdown) || conn != nil {
+		t.Fatal("bad: accept should return a shutdown error and a connection of nil value")
+	}
 	if err != nil && conn != nil {
-		t.Error("bad: accept should return a connection of nil value")
+		t.Fatal("bad: accept should return a connection of nil value")
```

Comment:
nbd but match original code and test both cases in one fell swoop

```suggestion
		t.Error("bad: accept should return a connection of nil value")
```